src/main/java/net/dv8tion/jda/api/interactions/commands/CommandInteraction.java

```diff
@@ -200,4 +203,54 @@ default OptionMapping getOption(@Nonnull String name)
         List<OptionMapping> options = getOptionsByName(name);
         return options.isEmpty() ? null : options.get(0);
     }
+
+    /**
+     * Gets the slash command String for this slash command.
+     * <br>This is similar to the String you see when clicking the interaction name in the client.
+     *
+     * <p>Example return for an echo command: {@code /say echo phrase: Say this}
+     *
+     * @return The command String for this slash command
+     */
+    @Nonnull
+    default String getCommandString()
+    {
+        //Get text like the text that appears when you hover over the interaction in discord
+        StringBuilder builder = new StringBuilder();
+        builder.append("/").append(getName());
+        if (getSubcommandGroup() != null)
+            builder.append(" ").append(getSubcommandGroup());
+        if (getSubcommandName() != null)
+            builder.append(" ").append(getSubcommandName());
+        for (OptionMapping o : getOptions())
+        {
+            builder.append(" ").append(o.getName()).append(": ");
+            switch (o.getType())
+            {
+            case CHANNEL:
+                builder.append("#").append(o.getAsGuildChannel().getName());
+                break;
+            case USER:
+                builder.append("@").append(o.getAsUser().getName());
+                break;
+            case ROLE:
+                builder.append("@").append(o.getAsRole().getName());
+                break;
+            case MENTIONABLE: //client only allows user or role mentionable as of Aug 4, 2021
+                if (o.getAsMentionable() instanceof RoleImpl)
+                    builder.append("@").append(((RoleImpl) o.getAsMentionable()).getName());
+                else if (o.getAsMentionable() instanceof MemberImpl)
+                    builder.append("@").append(((MemberImpl) o.getAsMentionable()).getEffectiveName());
+                else if (o.getAsMentionable() instanceof UserImpl)
+                    builder.append("@").append(((UserImpl) o.getAsMentionable()).getName());
```

Comment:
I doubt getEffectiveName makes sense, also you should probably just check the interface instead of concrete implementations and use the appropriate option getters instead of casting.

```suggestion
                if (o.getAsMentionable() instanceof Role)
                    builder.append("@").append(o.getAsRole().getName());
                else if (o.getAsMentionable() instanceof Member)
                    builder.append("@").append(o.getAsUser().getName());
                else if (o.getAsMentionable() instanceof User)
                    builder.append("@").append(o.getAsUser().getName());
```